publish-mqtt/src/main.rs

```diff
@@ -272,17 +334,47 @@ async fn connect_first_sensor_in_queue(
 }
 
 async fn connect_start_sensor<'a>(
-    bt_session: &'a BluetoothSession,
+    bt_session: MijiaSession,
     homie: &mut HomieDevice,
     sensor: &mut Sensor,
-) -> Result<(), Box<dyn Error>> {
-    let device = sensor.device(bt_session);
-    device.connect(CONNECT_TIMEOUT_MS)?;
-    start_notify_sensor(bt_session, &device)?;
-
-    homie.add_node(sensor.as_node()).await?;
-    sensor.last_update_timestamp = Instant::now();
-    Ok(())
+) -> Result<(), anyhow::Error> {
+    println!("Connecting from status: {:?}", sensor.connection_status);
+    bt_session
+        .connect(&sensor.object_path)
+        .await
+        .with_context(|| std::line!().to_string())?;
+    match start_notify_sensor(bt_session.clone(), &sensor.object_path).await {
```

Comment:
Could `start_notify_sensor` take a reference to bt_session instead, so you don't have to clone it?